// @react-aria/menu/dist/main.js
var $cSfnK$reactariautils = require("@react-aria/utils");
var $cSfnK$reactariainteractions = require("@react-aria/interactions");
var $cSfnK$reactariai18n = require("@react-aria/i18n");
var $cSfnK$reactariaoverlays = require("@react-aria/overlays");
var $cSfnK$reactariaselection = require("@react-aria/selection");
var $cSfnK$reactstatelycollections = require("@react-stately/collections");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $6ffbef18191cf50e$exports = {};

$parcel$export($6ffbef18191cf50e$exports, "useMenuTrigger", () => $6ffbef18191cf50e$export$dc9c12ed27dd1b49);
var $853781387242d7cb$exports = {};
var $f47f0979a691ab25$exports = {};
$f47f0979a691ab25$exports = JSON.parse("{\"longPressMessage\":\"اضغط مطولاً أو اضغط على Alt + السهم لأسفل لفتح القائمة\"}");


var $02a818d9662ba841$exports = {};
$02a818d9662ba841$exports = JSON.parse("{\"longPressMessage\":\"Натиснете продължително или натиснете Alt+ стрелка надолу, за да отворите менюто\"}");


var $b5ec5e34fd60606b$exports = {};
$b5ec5e34fd60606b$exports = JSON.parse("{\"longPressMessage\":\"Dlouhým stiskem nebo stisknutím kláves Alt + šipka dolů otevřete nabídku\"}");


var $96cf5b50416c94c5$exports = {};
$96cf5b50416c94c5$exports = JSON.parse("{\"longPressMessage\":\"Langt tryk eller tryk på Alt + pil ned for at åbne menuen\"}");


var $a0d65d51668f332f$exports = {};
$a0d65d51668f332f$exports = JSON.parse("{\"longPressMessage\":\"Drücken Sie lange oder drücken Sie Alt + Nach-unten, um das Menü zu öffnen\"}");


var $84e753c80b498a4a$exports = {};
$84e753c80b498a4a$exports = JSON.parse("{\"longPressMessage\":\"Πιέστε παρατεταμένα ή πατήστε Alt + κάτω βέλος για να ανοίξετε το μενού\"}");


var $d20766e7bd7d91d0$exports = {};
$d20766e7bd7d91d0$exports = JSON.parse("{\"longPressMessage\":\"Long press or press Alt + ArrowDown to open menu\"}");


var $2788c42a720bbc5b$exports = {};
$2788c42a720bbc5b$exports = JSON.parse("{\"longPressMessage\":\"Mantenga pulsado o pulse Alt + flecha abajo para abrir el menú\"}");


var $b4fc98cf53a50302$exports = {};
$b4fc98cf53a50302$exports = JSON.parse("{\"longPressMessage\":\"Menüü avamiseks vajutage pikalt või vajutage klahve Alt + allanool\"}");


var $324411c6468e6a44$exports = {};
$324411c6468e6a44$exports = JSON.parse("{\"longPressMessage\":\"Avaa valikko painamalla pohjassa tai näppäinyhdistelmällä Alt + Alanuoli\"}");


var $fe08364f6aba5fa8$exports = {};
$fe08364f6aba5fa8$exports = JSON.parse("{\"longPressMessage\":\"Appuyez de manière prolongée ou appuyez sur Alt + Flèche vers le bas pour ouvrir le menu.\"}");


var $429b0c5ee31de4c2$exports = {};
$429b0c5ee31de4c2$exports = JSON.parse("{\"longPressMessage\":\"לחץ לחיצה ארוכה או הקש Alt + ArrowDown כדי לפתוח את התפריט\"}");


var $c3974ce20cf32325$exports = {};
$c3974ce20cf32325$exports = JSON.parse("{\"longPressMessage\":\"Dugo pritisnite ili pritisnite Alt + strelicu prema dolje za otvaranje izbornika\"}");


var $8f89584581dae3a5$exports = {};
$8f89584581dae3a5$exports = JSON.parse("{\"longPressMessage\":\"Nyomja meg hosszan, vagy nyomja meg az Alt + lefele nyíl gombot a menü megnyitásához\"}");


var $deb131677e185eda$exports = {};
$deb131677e185eda$exports = JSON.parse("{\"longPressMessage\":\"Premere a lungo o premere Alt + Freccia giù per aprire il menu\"}");


var $1a672653ee657739$exports = {};
$1a672653ee657739$exports = JSON.parse("{\"longPressMessage\":\"長押しまたは Alt+下矢印キーでメニューを開く\"}");


var $3c2c600cb7cba19b$exports = {};
$3c2c600cb7cba19b$exports = JSON.parse("{\"longPressMessage\":\"길게 누르거나 Alt + 아래쪽 화살표를 눌러 메뉴 열기\"}");


var $1db0f589ec173b45$exports = {};
$1db0f589ec173b45$exports = JSON.parse("{\"longPressMessage\":\"Norėdami atidaryti meniu, nuspaudę palaikykite arba paspauskite „Alt + ArrowDown“.\"}");


var $85e6bf975d1094b1$exports = {};
$85e6bf975d1094b1$exports = JSON.parse("{\"longPressMessage\":\"Lai atvērtu izvēlni, turiet nospiestu vai nospiediet taustiņu kombināciju Alt + lejupvērstā bultiņa\"}");


var $5b30ea9def95deb7$exports = {};
$5b30ea9def95deb7$exports = JSON.parse("{\"longPressMessage\":\"Langt trykk eller trykk Alt + PilNed for å åpne menyen\"}");


var $36fc082ae9b07cd9$exports = {};
$36fc082ae9b07cd9$exports = JSON.parse("{\"longPressMessage\":\"Druk lang op Alt + pijl-omlaag of druk op Alt om het menu te openen\"}");


var $004ca257219ad5e5$exports = {};
$004ca257219ad5e5$exports = JSON.parse("{\"longPressMessage\":\"Naciśnij i przytrzymaj lub naciśnij klawisze Alt + Strzałka w dół, aby otworzyć menu\"}");


var $7185afda15938c1d$exports = {};
$7185afda15938c1d$exports = JSON.parse("{\"longPressMessage\":\"Pressione e segure ou pressione Alt + Seta para baixo para abrir o menu\"}");


var $f5ee82d2d2e387d1$exports = {};
$f5ee82d2d2e387d1$exports = JSON.parse("{\"longPressMessage\":\"Prima continuamente ou prima Alt + Seta Para Baixo para abrir o menu\"}");


var $9d73bbece95c54da$exports = {};
$9d73bbece95c54da$exports = JSON.parse("{\"longPressMessage\":\"Apăsați lung sau apăsați pe Alt + săgeată în jos pentru a deschide meniul\"}");


var $bc8566c135ef45e7$exports = {};
$bc8566c135ef45e7$exports = JSON.parse("{\"longPressMessage\":\"Нажмите и удерживайте или нажмите Alt + Стрелка вниз, чтобы открыть меню\"}");


var $505ac5577c8e9202$exports = {};
$505ac5577c8e9202$exports = JSON.parse("{\"longPressMessage\":\"Ponuku otvoríte dlhým stlačením alebo stlačením klávesu Alt + klávesu so šípkou nadol\"}");


var $4331407d77582ee6$exports = {};
$4331407d77582ee6$exports = JSON.parse("{\"longPressMessage\":\"Za odprtje menija pritisnite in držite gumb ali pritisnite Alt+puščica navzdol\"}");


var $98c11fa64e0657cd$exports = {};
$98c11fa64e0657cd$exports = JSON.parse("{\"longPressMessage\":\"Dugo pritisnite ili pritisnite Alt + strelicu prema dole da otvorite meni\"}");


var $1b7323508a8d2b62$exports = {};
$1b7323508a8d2b62$exports = JSON.parse("{\"longPressMessage\":\"Håll nedtryckt eller tryck på Alt + pil nedåt för att öppna menyn\"}");


var $61acf1b05216a392$exports = {};
$61acf1b05216a392$exports = JSON.parse("{\"longPressMessage\":\"Menüyü açmak için uzun basın veya Alt + Aşağı Ok tuşuna basın\"}");


var $ca0c4410f1dbdb12$exports = {};
$ca0c4410f1dbdb12$exports = JSON.parse("{\"longPressMessage\":\"Довго або звичайно натисніть комбінацію клавіш Alt і стрілка вниз, щоб відкрити меню\"}");


var $7870da185207c179$exports = {};
$7870da185207c179$exports = JSON.parse("{\"longPressMessage\":\"长按或按 Alt + 向下方向键以打开菜单\"}");


var $70d49959231766d5$exports = {};
$70d49959231766d5$exports = JSON.parse("{\"longPressMessage\":\"長按或按 Alt+向下鍵以開啟功能表\"}");


$853781387242d7cb$exports = {
    "ar-AE": $f47f0979a691ab25$exports,
    "bg-BG": $02a818d9662ba841$exports,
    "cs-CZ": $b5ec5e34fd60606b$exports,
    "da-DK": $96cf5b50416c94c5$exports,
    "de-DE": $a0d65d51668f332f$exports,
    "el-GR": $84e753c80b498a4a$exports,
    "en-US": $d20766e7bd7d91d0$exports,
    "es-ES": $2788c42a720bbc5b$exports,
    "et-EE": $b4fc98cf53a50302$exports,
    "fi-FI": $324411c6468e6a44$exports,
    "fr-FR": $fe08364f6aba5fa8$exports,
    "he-IL": $429b0c5ee31de4c2$exports,
    "hr-HR": $c3974ce20cf32325$exports,
    "hu-HU": $8f89584581dae3a5$exports,
    "it-IT": $deb131677e185eda$exports,
    "ja-JP": $1a672653ee657739$exports,
    "ko-KR": $3c2c600cb7cba19b$exports,
    "lt-LT": $1db0f589ec173b45$exports,
    "lv-LV": $85e6bf975d1094b1$exports,
    "nb-NO": $5b30ea9def95deb7$exports,
    "nl-NL": $36fc082ae9b07cd9$exports,
    "pl-PL": $004ca257219ad5e5$exports,
    "pt-BR": $7185afda15938c1d$exports,
    "pt-PT": $f5ee82d2d2e387d1$exports,
    "ro-RO": $9d73bbece95c54da$exports,
    "ru-RU": $bc8566c135ef45e7$exports,
    "sk-SK": $505ac5577c8e9202$exports,
    "sl-SI": $4331407d77582ee6$exports,
    "sr-SP": $98c11fa64e0657cd$exports,
    "sv-SE": $1b7323508a8d2b62$exports,
    "tr-TR": $61acf1b05216a392$exports,
    "uk-UA": $ca0c4410f1dbdb12$exports,
    "zh-CN": $7870da185207c179$exports,
    "zh-TW": $70d49959231766d5$exports
};






function $6ffbef18191cf50e$export$dc9c12ed27dd1b49(props, state, ref) {
    let { type: type = 'menu' , isDisabled: isDisabled , trigger: trigger = 'press'  } = props;
    let menuTriggerId = $cSfnK$reactariautils.useId();
    let { triggerProps: triggerProps , overlayProps: overlayProps  } = $cSfnK$reactariaoverlays.useOverlayTrigger({
        type: type
    }, state, ref);
    let onKeyDown = (e)=>{
        if (isDisabled) return;
        if (trigger === 'longPress' && !e.altKey) return;
        if (ref && ref.current) switch(e.key){
            case 'Enter':
            case ' ':
                if (trigger === 'longPress') return;
            // fallthrough
            case 'ArrowDown':
                // Stop propagation, unless it would already be handled by useKeyboard.
                if (!('continuePropagation' in e)) e.stopPropagation();
                e.preventDefault();
                state.toggle('first');
                break;
            case 'ArrowUp':
                if (!('continuePropagation' in e)) e.stopPropagation();
                e.preventDefault();
                state.toggle('last');
                break;
        }
    };
    let formatMessage = $cSfnK$reactariai18n.useMessageFormatter((/*@__PURE__*/$parcel$interopDefault($853781387242d7cb$exports)));
    let { longPressProps: longPressProps  } = $cSfnK$reactariainteractions.useLongPress({
        accessibilityDescription: formatMessage('longPressMessage'),
        onLongPressStart () {
            state.close();
        },
        onLongPress () {
            state.open('first');
        }
    });
    let pressProps = {
        onPressStart (e) {
            // For consistency with native, open the menu on mouse/key down, but touch up.
            if (e.pointerType !== 'touch' && e.pointerType !== 'keyboard' && !isDisabled) // If opened with a screen reader, auto focus the first item.
            // Otherwise, the menu itself will be focused.
            state.toggle(e.pointerType === 'virtual' ? 'first' : null);
        },
        onPress (e) {
            if (e.pointerType === 'touch' && !isDisabled) state.toggle();
        }
    };
    triggerProps = $cSfnK$reactariautils.mergeProps(triggerProps, trigger === 'press' ? pressProps : longPressProps);
    return {
        menuTriggerProps: {
            ...triggerProps,
            id: menuTriggerId,
            onKeyDown: onKeyDown
        },
        menuProps: {
            ...overlayProps,
            'aria-labelledby': menuTriggerId
        }
    };
}


var $e4bae9a72b1bc2fb$exports = {};

$parcel$export($e4bae9a72b1bc2fb$exports, "useMenu", () => $e4bae9a72b1bc2fb$export$38eaa17faae8f579);


function $e4bae9a72b1bc2fb$export$38eaa17faae8f579(props, state, ref) {
    let { shouldFocusWrap: shouldFocusWrap = true , ...otherProps } = props;
    if (!props['aria-label'] && !props['aria-labelledby']) console.warn('An aria-label or aria-labelledby prop is required for accessibility.');
    let domProps = $cSfnK$reactariautils.filterDOMProps(props, {
        labelable: true
    });
    let { listProps: listProps  } = $cSfnK$reactariaselection.useSelectableList({
        ...otherProps,
        ref: ref,
        selectionManager: state.selectionManager,
        collection: state.collection,
        disabledKeys: state.disabledKeys,
        shouldFocusWrap: shouldFocusWrap
    });
    return {
        menuProps: $cSfnK$reactariautils.mergeProps(domProps, {
            role: 'menu',
            ...listProps
        })
    };
}


var $d81c28283babe418$exports = {};

$parcel$export($d81c28283babe418$exports, "useMenuItem", () => $d81c28283babe418$export$9d32628fc2aea7da);




function $d81c28283babe418$export$9d32628fc2aea7da(props, state, ref) {
    let { isSelected: isSelected , isDisabled: isDisabled , key: key , onClose: onClose , closeOnSelect: closeOnSelect , isVirtualized: isVirtualized , onAction: onAction  } = props;
    let role = 'menuitem';
    if (state.selectionManager.selectionMode === 'single') role = 'menuitemradio';
    else if (state.selectionManager.selectionMode === 'multiple') role = 'menuitemcheckbox';
    let labelId = $cSfnK$reactariautils.useSlotId();
    let descriptionId = $cSfnK$reactariautils.useSlotId();
    let keyboardId = $cSfnK$reactariautils.useSlotId();
    let ariaProps = {
        'aria-disabled': isDisabled,
        role: role,
        'aria-label': props['aria-label'],
        'aria-labelledby': labelId,
        'aria-describedby': [
            descriptionId,
            keyboardId
        ].filter(Boolean).join(' ') || undefined
    };
    if (state.selectionManager.selectionMode !== 'none') ariaProps['aria-checked'] = isSelected;
    if (isVirtualized) {
        ariaProps['aria-posinset'] = state.collection.getItem(key).index;
        ariaProps['aria-setsize'] = $cSfnK$reactstatelycollections.getItemCount(state.collection);
    }
    let onKeyDown = (e)=>{
        // Ignore repeating events, which may have started on the menu trigger before moving
        // focus to the menu item. We want to wait for a second complete key press sequence.
        if (e.repeat) return;
        switch(e.key){
            case ' ':
                if (!isDisabled && state.selectionManager.selectionMode === 'none' && closeOnSelect !== false && onClose) onClose();
                break;
            case 'Enter':
                // The Enter key should always close on select, except if overridden.
                if (!isDisabled && closeOnSelect !== false && onClose) onClose();
                break;
        }
    };
    let onPressStart = (e)=>{
        if (e.pointerType === 'keyboard' && onAction) onAction(key);
    };
    let onPressUp = (e)=>{
        if (e.pointerType !== 'keyboard') {
            if (onAction) onAction(key);
            // Pressing a menu item should close by default in single selection mode but not multiple
            // selection mode, except if overridden by the closeOnSelect prop.
            if (onClose && (closeOnSelect !== null && closeOnSelect !== void 0 ? closeOnSelect : state.selectionManager.selectionMode !== 'multiple')) onClose();
        }
    };
    let { itemProps: itemProps  } = $cSfnK$reactariaselection.useSelectableItem({
        selectionManager: state.selectionManager,
        key: key,
        ref: ref,
        shouldSelectOnPressUp: true
    });
    let { pressProps: pressProps  } = $cSfnK$reactariainteractions.usePress({
        onPressStart: onPressStart,
        onPressUp: onPressUp,
        isDisabled: isDisabled
    });
    let { hoverProps: hoverProps  } = $cSfnK$reactariainteractions.useHover({
        isDisabled: isDisabled,
        onHoverStart () {
            if (!$cSfnK$reactariainteractions.isFocusVisible()) {
                state.selectionManager.setFocused(true);
                state.selectionManager.setFocusedKey(key);
            }
        }
    });
    return {
        menuItemProps: {
            ...ariaProps,
            ...$cSfnK$reactariautils.mergeProps(itemProps, pressProps, hoverProps, {
                onKeyDown: onKeyDown
            })
        },
        labelProps: {
            id: labelId
        },
        descriptionProps: {
            id: descriptionId
        },
        keyboardShortcutProps: {
            id: keyboardId
        }
    };
}


var $d1f36a3ac353172a$exports = {};

$parcel$export($d1f36a3ac353172a$exports, "useMenuSection", () => $d1f36a3ac353172a$export$73f7a44322579622);

function $d1f36a3ac353172a$export$73f7a44322579622(props) {
    let { heading: heading , 'aria-label': ariaLabel  } = props;
    let headingId = $cSfnK$reactariautils.useId();
    return {
        itemProps: {
            role: 'presentation'
        },
        headingProps: heading ? {
            // Techincally, menus cannot contain headings according to ARIA.
            // We hide the heading from assistive technology, and only use it
            // as a label for the nested group.
            id: headingId,
            'aria-hidden': true
        } : {
        },
        groupProps: {
            role: 'group',
            'aria-label': ariaLabel,
            'aria-labelledby': heading ? headingId : undefined
        }
    };
}


$parcel$exportWildcard(module.exports, $6ffbef18191cf50e$exports);
$parcel$exportWildcard(module.exports, $e4bae9a72b1bc2fb$exports);
$parcel$exportWildcard(module.exports, $d81c28283babe418$exports);
$parcel$exportWildcard(module.exports, $d1f36a3ac353172a$exports);
