// @react-aria/visually-hidden/dist/main.js
var $l9SOU$reactariautils = require("@react-aria/utils");
var $l9SOU$react = require("react");
var $l9SOU$reactariainteractions = require("@react-aria/interactions");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $8b518376223642b1$exports = {};

$parcel$export($8b518376223642b1$exports, "useVisuallyHidden", () => $8b518376223642b1$export$a966af930f325cab);
$parcel$export($8b518376223642b1$exports, "VisuallyHidden", () => $8b518376223642b1$export$439d29a4e110a164);



const $8b518376223642b1$var$styles = {
    border: 0,
    clip: 'rect(0 0 0 0)',
    clipPath: 'inset(50%)',
    height: 1,
    margin: '0 -1px -1px 0',
    overflow: 'hidden',
    padding: 0,
    position: 'absolute',
    width: 1,
    whiteSpace: 'nowrap'
};
function $8b518376223642b1$export$a966af930f325cab(props = {
}) {
    let { style: style , isFocusable: isFocusable  } = props;
    let [isFocused, setFocused] = $l9SOU$react.useState(false);
    let { focusProps: focusProps  } = $l9SOU$reactariainteractions.useFocus({
        isDisabled: !isFocusable,
        onFocusChange: setFocused
    });
    // If focused, don't hide the element.
    let combinedStyles = $l9SOU$react.useMemo(()=>{
        if (isFocused) return style;
        else if (style) return {
            ...$8b518376223642b1$var$styles,
            ...style
        };
        else return $8b518376223642b1$var$styles;
    }, [
        isFocused
    ]);
    return {
        visuallyHiddenProps: {
            ...focusProps,
            style: combinedStyles
        }
    };
}
function $8b518376223642b1$export$439d29a4e110a164(props) {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    let { children: children , elementType: Element = 'div' , isFocusable: isFocusable , style: style , ...otherProps } = props;
    let { visuallyHiddenProps: visuallyHiddenProps  } = $8b518376223642b1$export$a966af930f325cab(props);
    return(/*#__PURE__*/ ($parcel$interopDefault($l9SOU$react)).createElement(Element, $l9SOU$reactariautils.mergeProps(otherProps, visuallyHiddenProps), children));
}


$parcel$exportWildcard(module.exports, $8b518376223642b1$exports);
