// @react-aria/visually-hidden/dist/module.js
import {mergeProps as $hZMlr$mergeProps} from "@react-aria/utils";
import $hZMlr$react, {useState as $hZMlr$useState, useMemo as $hZMlr$useMemo} from "react";
import {useFocus as $hZMlr$useFocus} from "@react-aria/interactions";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $2b1a1e0abf5e6e14$exports = {};

$parcel$export($2b1a1e0abf5e6e14$exports, "useVisuallyHidden", () => $2b1a1e0abf5e6e14$export$a966af930f325cab);
$parcel$export($2b1a1e0abf5e6e14$exports, "VisuallyHidden", () => $2b1a1e0abf5e6e14$export$439d29a4e110a164);



const $2b1a1e0abf5e6e14$var$styles = {
    border: 0,
    clip: 'rect(0 0 0 0)',
    clipPath: 'inset(50%)',
    height: 1,
    margin: '0 -1px -1px 0',
    overflow: 'hidden',
    padding: 0,
    position: 'absolute',
    width: 1,
    whiteSpace: 'nowrap'
};
function $2b1a1e0abf5e6e14$export$a966af930f325cab(props = {
}) {
    let { style: style , isFocusable: isFocusable  } = props;
    let [isFocused, setFocused] = $hZMlr$useState(false);
    let { focusProps: focusProps  } = $hZMlr$useFocus({
        isDisabled: !isFocusable,
        onFocusChange: setFocused
    });
    // If focused, don't hide the element.
    let combinedStyles = $hZMlr$useMemo(()=>{
        if (isFocused) return style;
        else if (style) return {
            ...$2b1a1e0abf5e6e14$var$styles,
            ...style
        };
        else return $2b1a1e0abf5e6e14$var$styles;
    }, [
        isFocused
    ]);
    return {
        visuallyHiddenProps: {
            ...focusProps,
            style: combinedStyles
        }
    };
}
function $2b1a1e0abf5e6e14$export$439d29a4e110a164(props) {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    let { children: children , elementType: Element = 'div' , isFocusable: isFocusable , style: style , ...otherProps } = props;
    let { visuallyHiddenProps: visuallyHiddenProps  } = $2b1a1e0abf5e6e14$export$a966af930f325cab(props);
    return(/*#__PURE__*/ $hZMlr$react.createElement(Element, $hZMlr$mergeProps(otherProps, visuallyHiddenProps), children));
}




export {$2b1a1e0abf5e6e14$export$a966af930f325cab as useVisuallyHidden, $2b1a1e0abf5e6e14$export$439d29a4e110a164 as VisuallyHidden};
