// @react-stately/checkbox/dist/module.js
import {useControlledState as $bk5pn$useControlledState} from "@react-stately/utils";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $a67f6fc5ad4ce17c$exports = {};

$parcel$export($a67f6fc5ad4ce17c$exports, "useCheckboxGroupState", () => $a67f6fc5ad4ce17c$export$daff6da51032a415);

function $a67f6fc5ad4ce17c$export$daff6da51032a415(props = {
}) {
    let [selectedValues, setValue] = $bk5pn$useControlledState(props.value, props.defaultValue || [], props.onChange);
    const state = {
        value: selectedValues,
        setValue (value) {
            if (props.isReadOnly || props.isDisabled) return;
            setValue(value);
        },
        isDisabled: props.isDisabled || false,
        isReadOnly: props.isReadOnly || false,
        isSelected (value) {
            return selectedValues.includes(value);
        },
        addValue (value) {
            if (props.isReadOnly || props.isDisabled) return;
            if (!selectedValues.includes(value)) setValue(selectedValues.concat(value));
        },
        removeValue (value) {
            if (props.isReadOnly || props.isDisabled) return;
            if (selectedValues.includes(value)) setValue(selectedValues.filter((existingValue)=>existingValue !== value
            ));
        },
        toggleValue (value) {
            if (props.isReadOnly || props.isDisabled) return;
            if (selectedValues.includes(value)) setValue(selectedValues.filter((existingValue)=>existingValue !== value
            ));
            else setValue(selectedValues.concat(value));
        }
    };
    return state;
}




export {$a67f6fc5ad4ce17c$export$daff6da51032a415 as useCheckboxGroupState};
