// @react-aria/checkbox/dist/main.js
var $g1NuS$react = require("react");
var $g1NuS$reactariatoggle = require("@react-aria/toggle");
var $g1NuS$reactariautils = require("@react-aria/utils");
var $g1NuS$reactarialabel = require("@react-aria/label");
var $g1NuS$reactstatelytoggle = require("@react-stately/toggle");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $0ec554106b85b62e$exports = {};

$parcel$export($0ec554106b85b62e$exports, "useCheckbox", () => $0ec554106b85b62e$export$e375f10ce42261c5);


function $0ec554106b85b62e$export$e375f10ce42261c5(props, state, inputRef) {
    let { inputProps: inputProps  } = $g1NuS$reactariatoggle.useToggle(props, state, inputRef);
    let { isSelected: isSelected  } = state;
    let { isIndeterminate: isIndeterminate  } = props;
    $g1NuS$react.useEffect(()=>{
        // indeterminate is a property, but it can only be set via javascript
        // https://css-tricks.com/indeterminate-checkboxes/
        if (inputRef.current) inputRef.current.indeterminate = isIndeterminate;
    });
    return {
        inputProps: {
            ...inputProps,
            checked: isSelected,
            'aria-checked': isIndeterminate ? 'mixed' : isSelected
        }
    };
}


var $0505cbc377dd66f2$exports = {};

$parcel$export($0505cbc377dd66f2$exports, "useCheckboxGroup", () => $0505cbc377dd66f2$export$49ff6f28c54f1cbe);
const $8b3b549de2099acb$export$31440636951aa68c = new WeakMap();




function $0505cbc377dd66f2$export$49ff6f28c54f1cbe(props, state) {
    let { isDisabled: isDisabled , name: name  } = props;
    let { labelProps: labelProps , fieldProps: fieldProps  } = $g1NuS$reactarialabel.useLabel({
        ...props,
        // Checkbox group is not an HTML input element so it
        // shouldn't be labeled by a <label> element.
        labelElementType: 'span'
    });
    let domProps = $g1NuS$reactariautils.filterDOMProps(props, {
        labelable: true
    });
    // Pass name prop from group to all items by attaching to the state.
    $8b3b549de2099acb$export$31440636951aa68c.set(state, name);
    return {
        groupProps: $g1NuS$reactariautils.mergeProps(domProps, {
            role: 'group',
            'aria-disabled': isDisabled || undefined,
            ...fieldProps
        }),
        labelProps: labelProps
    };
}


var $2c160cc84b857a22$exports = {};

$parcel$export($2c160cc84b857a22$exports, "useCheckboxGroupItem", () => $2c160cc84b857a22$export$353b32fc6898d37d);



function $2c160cc84b857a22$export$353b32fc6898d37d(props, state, inputRef) {
    const toggleState = $g1NuS$reactstatelytoggle.useToggleState({
        isReadOnly: props.isReadOnly || state.isReadOnly,
        isSelected: state.isSelected(props.value),
        onChange (isSelected) {
            if (isSelected) state.addValue(props.value);
            else state.removeValue(props.value);
            if (props.onChange) props.onChange(isSelected);
        }
    });
    let { inputProps: inputProps  } = $0ec554106b85b62e$export$e375f10ce42261c5({
        ...props,
        isReadOnly: props.isReadOnly || state.isReadOnly,
        isDisabled: props.isDisabled || state.isDisabled,
        name: props.name || $8b3b549de2099acb$export$31440636951aa68c.get(state)
    }, toggleState, inputRef);
    return {
        inputProps: inputProps
    };
}


$parcel$exportWildcard(module.exports, $0ec554106b85b62e$exports);
$parcel$exportWildcard(module.exports, $0505cbc377dd66f2$exports);
$parcel$exportWildcard(module.exports, $2c160cc84b857a22$exports);
