// @react-stately/overlays/dist/main.js
var $lubP4$reactstatelyutils = require("@react-stately/utils");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $db9a34e644d75b32$exports = {};

$parcel$export($db9a34e644d75b32$exports, "useOverlayTriggerState", () => $db9a34e644d75b32$export$61c6a8c84e605fb6);

function $db9a34e644d75b32$export$61c6a8c84e605fb6(props) {
    let [isOpen, setOpen] = $lubP4$reactstatelyutils.useControlledState(props.isOpen, props.defaultOpen || false, props.onOpenChange);
    return {
        isOpen: isOpen,
        open () {
            setOpen(true);
        },
        close () {
            setOpen(false);
        },
        toggle () {
            setOpen(!isOpen);
        }
    };
}


$parcel$exportWildcard(module.exports, $db9a34e644d75b32$exports);
