// @react-aria/listbox/dist/main.js
var $2DMzk$reactariautils = require("@react-aria/utils");
var $2DMzk$reactariainteractions = require("@react-aria/interactions");
var $2DMzk$reactarialabel = require("@react-aria/label");
var $2DMzk$reactariaselection = require("@react-aria/selection");
var $2DMzk$reactstatelycollections = require("@react-stately/collections");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $50e07b4958ef8269$exports = {};

$parcel$export($50e07b4958ef8269$exports, "useListBox", () => $50e07b4958ef8269$export$50eacbbf140a3141);

var $29f6a4f44d3a6168$exports = {};

$parcel$export($29f6a4f44d3a6168$exports, "listData", () => $29f6a4f44d3a6168$export$3585ede4d035bf14);
$parcel$export($29f6a4f44d3a6168$exports, "getItemId", () => $29f6a4f44d3a6168$export$9145995848b05025);
const $29f6a4f44d3a6168$export$3585ede4d035bf14 = new WeakMap();
function $29f6a4f44d3a6168$var$normalizeKey(key) {
    if (typeof key === 'string') return key.replace(/\s*/g, '');
    return '' + key;
}
function $29f6a4f44d3a6168$export$9145995848b05025(state, itemKey) {
    let data = $29f6a4f44d3a6168$export$3585ede4d035bf14.get(state);
    if (!data) throw new Error('Unknown list');
    return `${data.id}-option-${$29f6a4f44d3a6168$var$normalizeKey(itemKey)}`;
}






function $50e07b4958ef8269$export$50eacbbf140a3141(props, state, ref) {
    let domProps = $2DMzk$reactariautils.filterDOMProps(props, {
        labelable: true
    });
    let { listProps: listProps  } = $2DMzk$reactariaselection.useSelectableList({
        ...props,
        ref: ref,
        selectionManager: state.selectionManager,
        collection: state.collection,
        disabledKeys: state.disabledKeys
    });
    let { focusWithinProps: focusWithinProps  } = $2DMzk$reactariainteractions.useFocusWithin({
        onFocusWithin: props.onFocus,
        onBlurWithin: props.onBlur,
        onFocusWithinChange: props.onFocusChange
    });
    // Share list id and some props with child options.
    let id = $2DMzk$reactariautils.useId(props.id);
    $29f6a4f44d3a6168$export$3585ede4d035bf14.set(state, {
        id: id,
        shouldUseVirtualFocus: props.shouldUseVirtualFocus,
        shouldSelectOnPressUp: props.shouldSelectOnPressUp,
        shouldFocusOnHover: props.shouldFocusOnHover,
        isVirtualized: props.isVirtualized
    });
    let { labelProps: labelProps , fieldProps: fieldProps  } = $2DMzk$reactarialabel.useLabel({
        ...props,
        id: id,
        // listbox is not an HTML input element so it
        // shouldn't be labeled by a <label> element.
        labelElementType: 'span'
    });
    return {
        labelProps: labelProps,
        listBoxProps: $2DMzk$reactariautils.mergeProps(domProps, focusWithinProps, state.selectionManager.selectionMode === 'multiple' ? {
            'aria-multiselectable': 'true'
        } : {
        }, {
            role: 'listbox',
            ...$2DMzk$reactariautils.mergeProps(fieldProps, listProps)
        })
    };
}


var $29933ad53619f82f$exports = {};

$parcel$export($29933ad53619f82f$exports, "useOption", () => $29933ad53619f82f$export$497855f14858aa34);





function $29933ad53619f82f$export$497855f14858aa34(props, state, ref) {
    let { key: key  } = props;
    let data = $29f6a4f44d3a6168$export$3585ede4d035bf14.get(state);
    var _isDisabled;
    let isDisabled = (_isDisabled = props.isDisabled) !== null && _isDisabled !== void 0 ? _isDisabled : state.disabledKeys.has(key);
    var _isSelected;
    let isSelected = (_isSelected = props.isSelected) !== null && _isSelected !== void 0 ? _isSelected : state.selectionManager.isSelected(key);
    let isFocused = state.selectionManager.focusedKey === key;
    var _shouldSelectOnPressUp;
    let shouldSelectOnPressUp = (_shouldSelectOnPressUp = props.shouldSelectOnPressUp) !== null && _shouldSelectOnPressUp !== void 0 ? _shouldSelectOnPressUp : data.shouldSelectOnPressUp;
    var _shouldFocusOnHover;
    let shouldFocusOnHover = (_shouldFocusOnHover = props.shouldFocusOnHover) !== null && _shouldFocusOnHover !== void 0 ? _shouldFocusOnHover : data.shouldFocusOnHover;
    var _shouldUseVirtualFocus;
    let shouldUseVirtualFocus = (_shouldUseVirtualFocus = props.shouldUseVirtualFocus) !== null && _shouldUseVirtualFocus !== void 0 ? _shouldUseVirtualFocus : data.shouldUseVirtualFocus;
    var _isVirtualized;
    let isVirtualized = (_isVirtualized = props.isVirtualized) !== null && _isVirtualized !== void 0 ? _isVirtualized : data.isVirtualized;
    let labelId = $2DMzk$reactariautils.useSlotId();
    let descriptionId = $2DMzk$reactariautils.useSlotId();
    let optionProps = {
        role: 'option',
        'aria-disabled': isDisabled,
        'aria-selected': state.selectionManager.selectionMode !== 'none' ? isSelected : undefined
    };
    // Safari with VoiceOver on macOS misreads options with aria-labelledby or aria-label as simply "text".
    // We should not map slots to the label and description on Safari and instead just have VoiceOver read the textContent.
    // https://bugs.webkit.org/show_bug.cgi?id=209279
    if (!($2DMzk$reactariautils.isMac() && $2DMzk$reactariautils.isWebKit())) {
        optionProps['aria-label'] = props['aria-label'];
        optionProps['aria-labelledby'] = labelId;
        optionProps['aria-describedby'] = descriptionId;
    }
    if (isVirtualized) {
        optionProps['aria-posinset'] = state.collection.getItem(key).index + 1;
        optionProps['aria-setsize'] = $2DMzk$reactstatelycollections.getItemCount(state.collection);
    }
    let { itemProps: itemProps , isPressed: isPressed  } = $2DMzk$reactariaselection.useSelectableItem({
        selectionManager: state.selectionManager,
        key: key,
        ref: ref,
        shouldSelectOnPressUp: shouldSelectOnPressUp,
        isVirtualized: isVirtualized,
        shouldUseVirtualFocus: shouldUseVirtualFocus,
        isDisabled: isDisabled
    });
    let { hoverProps: hoverProps  } = $2DMzk$reactariainteractions.useHover({
        isDisabled: isDisabled || !shouldFocusOnHover,
        onHoverStart () {
            if (!$2DMzk$reactariainteractions.isFocusVisible()) {
                state.selectionManager.setFocused(true);
                state.selectionManager.setFocusedKey(key);
            }
        }
    });
    return {
        optionProps: {
            ...optionProps,
            ...$2DMzk$reactariautils.mergeProps(itemProps, hoverProps),
            id: $29f6a4f44d3a6168$export$9145995848b05025(state, key)
        },
        labelProps: {
            id: labelId
        },
        descriptionProps: {
            id: descriptionId
        },
        isFocused: isFocused,
        isSelected: isSelected,
        isDisabled: isDisabled,
        isPressed: isPressed
    };
}


var $8262af1bc01e22eb$exports = {};

$parcel$export($8262af1bc01e22eb$exports, "useListBoxSection", () => $8262af1bc01e22eb$export$c3f9f39876e4bc7);

function $8262af1bc01e22eb$export$c3f9f39876e4bc7(props) {
    let { heading: heading , 'aria-label': ariaLabel  } = props;
    let headingId = $2DMzk$reactariautils.useId();
    return {
        itemProps: {
            role: 'presentation'
        },
        headingProps: heading ? {
            // Techincally, listbox cannot contain headings according to ARIA.
            // We hide the heading from assistive technology, and only use it
            // as a label for the nested group.
            id: headingId,
            'aria-hidden': true
        } : {
        },
        groupProps: {
            role: 'group',
            'aria-label': ariaLabel,
            'aria-labelledby': heading ? headingId : undefined
        }
    };
}



$parcel$exportWildcard(module.exports, $50e07b4958ef8269$exports);
$parcel$exportWildcard(module.exports, $29933ad53619f82f$exports);
$parcel$exportWildcard(module.exports, $8262af1bc01e22eb$exports);
$parcel$exportWildcard(module.exports, $29f6a4f44d3a6168$exports);
